internal/provider/jwt_issuers_test.go

```diff
@@ -0,0 +1,364 @@
+				Destroy:           true,
+			},
+		},
+		CheckDestroy: testJWTIssuerDestroy(resourceNameTest),
+	})
+}
+
+func getTestJWTIssuerCreateConfig(issuerURL string) string {
+	identityMapString := "[\n"
+	for _, identityMapEntry := range testIdentityMap {
+		identityMapString += "{\n"
+		identityMapString += fmt.Sprintf("token_identity = %s\n", identityMapEntry.TokenIdentity)
+		identityMapString += fmt.Sprintf("cc_identity = %s\n", identityMapEntry.CcIdentity)
+		identityMapString += fmt.Sprintf("is_regex = %s\n", identityMapEntry.IsRegex)
+		identityMapString += "},\n"
+	}
+	identityMapString += "]"
+
+	return fmt.Sprintf(`
+resource "cockroach_jwt_issuer" "test" {
+  issuer_url = "%s"
+  audience = "%s"
+  jwks = "%s"
+  claim = "%s"
+  identity_map = %s
+}
+`, issuerURL, testAudience, testJWKS, testClaim, identityMapString)
+}
+
+func getTestJWTIssuerUpdateConfig(issuerURL string) string {
+	identityMapString := "[\n"
+	for _, identityMapEntry := range testIdentityMapUpdated {
+		identityMapString += "{\n"
+		identityMapString += fmt.Sprintf("token_identity = %s\n", identityMapEntry.TokenIdentity)
+		identityMapString += fmt.Sprintf("cc_identity = %s\n", identityMapEntry.CcIdentity)
+		identityMapString += fmt.Sprintf("is_regex = %s\n", identityMapEntry.IsRegex)
+		identityMapString += "},\n"
+	}
+	identityMapString += "]"
+
+	return fmt.Sprintf(`
+resource "cockroach_jwt_issuer" "test" {
+  issuer_url = "%s"
+  audience = "%s"
+  jwks = "%s"
+  claim = "%s"
+  identity_map = %s
+}
+`, issuerURL, testAudienceUpdated, testJWKSUpdated, testClaimUpdated, identityMapString)
+}
+
+func testJWTIssuer(resourceName, issuerURL string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		p := testAccProvider.(*provider)
+		p.service = NewService(cl)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		traceAPICall("GetJWTIssuer")
+		jwtIssuerResp, _, err := p.service.GetJWTIssuer(context.TODO(), rs.Primary.ID)
+		if err == nil {
+			respIdentityMap := *jwtIssuerResp.IdentityMap
+			if jwtIssuerResp.IssuerUrl == issuerURL &&
+				jwtIssuerResp.Audience == testAudience &&
+				*jwtIssuerResp.Jwks == testJWKS &&
+				*jwtIssuerResp.Claim == testClaim &&
+				len(respIdentityMap) == 2 &&
+				*respIdentityMap[0].TokenIdentity == testIdentityMap[0].TokenIdentity.ValueString() &&
+				*respIdentityMap[0].CcIdentity == testIdentityMap[0].CcIdentity.ValueString() &&
+				*respIdentityMap[0].IsRegex == testIdentityMap[0].IsRegex.ValueBool() &&
+				*respIdentityMap[1].TokenIdentity == testIdentityMap[1].TokenIdentity.ValueString() &&
+				*respIdentityMap[1].CcIdentity == testIdentityMap[1].CcIdentity.ValueString() &&
+				*respIdentityMap[1].IsRegex == testIdentityMap[1].IsRegex.ValueBool() {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("JWT Issuer configuration does not have correct values")
```

Comment:
It might be nice to include the 2 objects for comparison in this error message.  It looks like we use testify in some places.  I guess its not exactly in the right format but if you could convert one of the object it might be nicer to use something like `require.Equal(t, ...)`

Reply:
Updated. Thank you!
Did not use `require.Equal(t, ...)` to let the error be reported as per the semantics of `TestCheckFunc`.

Reply:
cool.  Looks good.